packages/client/tiptap/extensions/database/data.ts

```diff
@@ -0,0 +1,120 @@
+import * as Y from 'yjs'
+import {DataType} from './types'
+
+export type ColumnId = string
+export type RowId = string
+export type ColumnMeta = {
+  name: string
+  type: DataType
+}
+
+const cloneColumnMeta = (meta: ColumnMeta): ColumnMeta => {
+  return {
+    name: meta.name,
+    type: meta.type
+  }
+}
+
+const generateId = (doc: Y.Doc) => {
+  const id = Math.random().toString(36).substring(2, 9)
```

Comment:
-1 you're gonna hate me for this... by the time we get to 100k rows, each insert has a 6.2% chance of collision, so making sure we don't have a collision is the right thing to do! ...but if we handle a large paste, like 100k rows, a O(n) could start taking a reeeeeal long time & be a tricky bug to find since it'd lock up the process. 

couple ideas:
- crypto.generateUUID (agreed, it'd be big & ugly)
- make a snowflake using yjs clientID + timestamp partial
- keep all the rows in a YJS Array & ditch IDs for rows